src/js/videojs.wavesurfer.js

```diff
@@ -253,13 +253,31 @@ class Wavesurfer extends Plugin {
      * @param {string|blob|file} url - Either the URL of the audio file,
      *     a Blob or a File object.
      */
-    load(url) {
-        if (url instanceof Blob || url instanceof File) {
-            this.log('Loading object: ' + JSON.stringify(url));
-            this.surfer.loadBlob(url);
+    load(audioUrl, peakUrl) {
+        let player = this;
+        if (audioUrl instanceof Blob || audioUrl instanceof File) {
+            player.log('Loading object: ' + JSON.stringify(audioUrl));
+            player.surfer.loadBlob(audioUrl);
         } else {
-            this.log('Loading URL: ' + url);
-            this.surfer.load(url);
+            player.log('Loading URL: ' + audioUrl + '\nLoading Peak Data URL: ' + peakUrl);
+            // Load peak data from file
+            if (peakUrl !== undefined) {
```

Comment:
there's no need to use the `XMLHttpRequest` here directly, use wavesurfer's https://github.com/katspaugh/wavesurfer.js/blob/master/src/util/ajax.js `ajax` util instead.